group-generators/helpers/data-providers/gitpoap/index.ts

```diff
@@ -0,0 +1,84 @@
+import axios from "axios";
+import { GitPoap, GitPoapAddresses } from "./types";
+import { FetchedData } from "topics/group";
+
+
+// fixes https://github.com/sismo-core/sismo-hub/issues/1454
+// api docs https://docs.gitpoap.io/api
+
+export class GitPoapProvider {
+  url: string;
+
+
+  public constructor() {
+    this.url = "https://public-api.gitpoap.io/";
+  }
+
+  async getGitPoap(endpoint: string): Promise<any> {
+    const { data: res } = await axios({
+      url: this.url + endpoint, 
+      method: "get",
+    });
+    return res;
+  }
+
+
+  public async getGitPoapHoldersByEventId(gitPoapEventId: string ): Promise<FetchedData> {
+    const dataProfiles: FetchedData = {};
+    let holdersAddress: string[];
+    try {
+      const req: GitPoapAddresses= await this.getGitPoap("/v1/gitpoaps/" + gitPoapEventId + "/addresses"); 
+      holdersAddress = req.addresses;
+    } catch (error) {
+      throw new Error("Error fetching total number of users: " + error);
+    }
+
+    await Promise.all(holdersAddress)
+    .then((addresses) => {
+      addresses.forEach((address) => {
+        if (address != "") {
+          dataProfiles[address] = 1;
+        }
+      });
+    });
+
+    return dataProfiles;
+  }
+
+  public async getGitPoapHoldersByEventIdCount(gitPoapEventId: string): Promise<number> {
+    const  holders = this.getGitPoapHoldersByEventId(gitPoapEventId);
+    return Object(holders).keys().length;
+  }
+
+  public async getGitPoapEventsIDByAddress(address: string ): Promise<FetchedData> {
```

Comment:
the function here is not really use full because it fetch all the gitpoap ids for an address and returns an empty FetchData object.
And you can't make a group with gitpoap if, you can only create group with web2 or web3 accounts.
So I don't see the utility of this function

Reply:
removed with https://github.com/sismo-core/sismo-hub/pull/1555/commits/39b74620ee76b81113e37b7f3426f728d3f9d406